### DISCUSSION TITLE
How are you editing/creating files & folders?

### DISCUSSION THREAD

**Discussion by GollyJer** [custom-components/pyscript#507] [Q&A] [answered]
My setup is:
Unraid server running the HA Docker image.
---------------

What is the best way to add/edit files? Ideally, I'm looking to use VSCode natively on my local machine.

So far I've been able to 
- Connect to the Unraid server using the `Remote SSH` extension.
- Connect to the HA container using the `Docker` extension.
- Edit files by opening and saving them.  
<img width="600" alt="SCR-20230806-rgdd" src="https://github.com/custom-components/pyscript/assets/689204/9e6b163a-54fe-44d2-b044-2dc318ddc100">

This is not terrible, but I can only view/edit this way. I'm unable to create new files/folders.
How are you guys approaching script dev?

Thanks!

**Comment by oskretc** [comment]
I'm using code-server (vscode in the browser) installed in a separate machine (unraid) and sharing the config folder with samba. then I can edit using vscode in anymachine and access to terminal as well

**Comment by ALERTua** [comment]
Unraid HA VM, Samba HA plugin, bound /config to my Windows as a drive letter, opened it as a project in PyCharm and created a local virtual environment for code highlighting. voila.

**Reply by GollyJer** [reply]
Things would be easier if I could get the Unraid VM working. 
For some reason, HA doesn't pick up a network connection when it fires up in VM. I had to go the Docker route.

**Reply by ALERTua** [reply]
Try these network settings for your VM.
![image](https://github.com/custom-components/pyscript/assets/8375129/08002b9d-b7db-4df5-9dc7-239feb23ff87)

**Reply by GollyJer** [reply]
Hey. Thanks for the help! 
Sadly that's how it's already set up. 

If you run `network info` in the HA CLI do you anything in the `ipv4.nameservers` section?
Everything else looks normal to but HA never gets past the Preparing Home Assistant screen.

<img width="276" alt="SCR-20230807-krle" src="https://github.com/custom-components/pyscript/assets/689204/f8234b78-56a7-4284-88ea-792026f0893d">

**Reply by ALERTua** [reply]
`ipv4.nameservers` has both my primary and secondary network DNS.

**Reply by GollyJer** [reply]
Couldn't for the life of me get it to pick up nameservers so I moved forward with figuring out how to connect VS Code to HA Core. Answer posted here as well.

**Comment by GollyJer** [comment, marked as answer]
I solved this without Samba and shared the solution on StackOverflow for easy access.

[How can I connect VS Code to a Home Assistant docker container running on Unraid Server?](https://stackoverflow.com/questions/76864400/how-can-i-connect-vs-code-to-a-home-assistant-docker-container-running-on-unraid)
